tests/failpoints_tests.rs

```diff
@@ -0,0 +1,64 @@
+#![cfg(feature = "integration-tests")]
```

Comment:
Why is it seperated from `tests/integration_tests.rs`?

Reply:
This test have failpoints.

Reply:
I suggest to put them together

Reply:
It follows the suggestions given by `fail-rs`.

> Carefully consider complex, concurrent, non-deterministic combinations of fail points. Put test cases exercising fail points into their own test crate.